### PR TITLE
SmallRye Reactive Messaging 2.0

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningMessages.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private AtomicInteger count = new AtomicInteger();
+    private List<String> threads = new CopyOnWriteArrayList<>();
+
+    @Blocking
+    @Outgoing("infinite-producer-msg")
+    public Message<Integer> create() {
+        try {
+            Thread.sleep(200);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningMessages.java:25]: Can you add: Thread.currentThread().interrupt();

Comment (cescoffier) [reply extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningMessages.java:25]: (and remove the dump of the stack trace maybe)

File: extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningPayloads.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private AtomicInteger count = new AtomicInteger();
+    private List<String> threads = new CopyOnWriteArrayList<>();
+
+    @Blocking
+    @Outgoing("infinite-producer-payload")
+    public int create() {
+        try {
+            Thread.sleep(200);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningPayloads.java:24]: Same comment as above (reset interrupt flag)

**Comment by gsmet** [comment]
@michalszynkiewicz @cescoffier is it something for 1.4 or for 1.5?

File: extensions/smallrye-reactive-messaging/runtime/pom.xml [hunk 3]
```diff
@@ -25,6 +25,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-vertx-core</artifactId>
         </dependency>
+        <dependency>
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging/runtime/pom.xml:28]: We should be able to remove the quarkus-vertx-core dependency, as it's a transitive dependency of this one.

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java [hunk 4]
```diff
@@ -44,6 +44,12 @@
 
     private Class<? extends Invoker> invokerClass;
 
+    private boolean blocking;
+
+    private boolean blockingExecutionOrdered;
+
+    String workerPoolName;
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java:51]: Should it be private?

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import io.smallrye.reactive.messaging.connectors.WorkerPoolRegistry;
+import io.smallrye.reactive.messaging.helpers.Validation;
+import io.vertx.core.Handler;
+import io.vertx.mutiny.core.Promise;
+import io.vertx.mutiny.core.WorkerExecutor;
+
+@AlternativePriority(1)
+@ApplicationScoped
+// TODO: create a different entry for WorkerPoolRegistry than `analyzeWorker` and drop this class
+public class QuarkusWorkerPoolRegistry extends WorkerPoolRegistry {
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java:32]: IS the only difference with the upstream version the AlternativePriority?

**Review by cescoffier — CHANGES_REQUESTED** [review]
Great work! Mostly good!
I've written a few comments, but nothing important.

**Comment by cescoffier** [comment]
@gsmet I would prefer 1.4, but that's your call.

File: extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningMessages.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningMessages.java:25]: done

File: extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningPayloads.java (hunk 2 above)
Comment (michalszynkiewicz) [reply extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/blocking/beans/BeanReturningPayloads.java:24]: done

File: extensions/smallrye-reactive-messaging/runtime/pom.xml (hunk 3 above)
Comment (michalszynkiewicz) [reply extensions/smallrye-reactive-messaging/runtime/pom.xml:28]: done, also added dependency between deployments

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java (hunk 4 above)
Comment (michalszynkiewicz) [reply extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java:51]: done

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java (hunk 5 above)
Comment (michalszynkiewicz) [reply extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java:32]: the difference is the `defineWorker` method; the upstream version only works on `AnnotatedType`: https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/connectors/WorkerPoolRegistry.java#L98

**Comment by gsmet** [comment]
Fine by me if we can get this merged before 6PM Paris time today.

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java (hunk 5 above)
Comment (cescoffier) [reply extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java:32]: Thanks!

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
The JAva 14 CI issue is unrelated (it's the other AMQP issue I need to fix...)

**Pull request merged**